packages/connection-form/src/components/advanced-options-tabs/csfle-tab/encrypted-field-config-input.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import type { Document } from 'mongodb';
+import {
+  encryptedFieldConfigToText,
+  textToEncryptedFieldConfig,
+} from '../../../utils/csfle-handler';
+import { Label } from '@mongodb-js/compass-components';
+
+import 'ace-builds';
+import AceEditor from 'react-ace';
+import 'ace-builds/src-noconflict/ext-language_tools';
+import 'mongodb-ace-mode';
+import 'mongodb-ace-theme';
+
+/**
+ * Options for the ACE editor.
+ */
+const OPTIONS = {
+  enableLiveAutocompletion: false,
+  tabSize: 2,
+  fontSize: 11,
+  minLines: 10,
+  maxLines: Infinity,
+  showGutter: true,
+  useWorker: false,
+  mode: 'ace/mode/mongodb',
+};
+
+function EncryptedFieldConfigInput({
+  encryptedFieldConfig,
+  errorMessage,
+  onChange,
+}: {
+  encryptedFieldConfig: Document | undefined;
+  errorMessage: string | undefined;
+  onChange: (value: Document | undefined) => void;
+}): React.ReactElement {
+  function changeCurrentText(newText: string) {
+    if (!newText.trim()) {
+      onChange(undefined);
+      return;
+    }
+
+    onChange(textToEncryptedFieldConfig(newText));
+  }
+
+  const currentText = encryptedFieldConfig
+    ? encryptedFieldConfigToText(encryptedFieldConfig)
+    : '';
+
+  return (
+    <div>
+      <Label htmlFor="TODO">EncryptedFieldConfig (recommended)</Label>
+      <AceEditor
+        mode="javascript" // will be set to mongodb as part of OPTIONS
+        theme="mongodb"
+        width="100%"
+        value={currentText}
+        onChange={changeCurrentText}
+        editorProps={{ $blockScrolling: Infinity }}
+        name="import-pipeline-editor"
+        setOptions={OPTIONS}
+      />
+      {
+        /* TODO: proper error message display! */ errorMessage && (
```

Comment:
@Anemy Any ideas for doing this? Do we have some easy way to display error messages for ace editor inputs?

Reply:
Could the errors at the bottom of the form be an option - is there a reason we'd like to keep these errors shown more locally?

Reply:
@Anemy It's an option I guess, yeah -- it's a bit inconsistent with how we use the `errorMessage` prop of TextInput/TextArea fields where we display it immediately next to the fields, but it doesn't look like leafygreen exposes that specific part as a component anywhere